src/java/us/kbase/templates/module_test_perl_client.vm.properties

```diff
@@ -33,25 +32,43 @@ my $auth_token   = Bio::KBase::AuthToken->new(
 my $ctx = LocalCallContext->new( $token, $auth_token->user_id );
 $${module_name}::${module_name}Server::CallContext = $ctx;
 
-my $impl = ${module_name}::${module_name}Impl->new();
+sub get_ws_client {
+
+    my $ws_url  = $config->{ "workspace-url" };
+    $ws_client  //= installed_clients::WorkspaceClient->new( $ws_url, token => $token );
```

Comment:
So this "//=" is a conditional assignment and not a comment?

Reply:
Yup -- it means "If `$ws_client` is not defined, set it to `installed_clients::WorkspaceClient->new( $ws_url, token => $token );`".

You could also write

```perl
$ws_client = installed_clients::WorkspaceClient->new( $ws_url, token => $token ) unless defined $ws_client;
```